src/sig/sig.h

```diff
@@ -0,0 +1,138 @@
+/**
+ * \file sig.h
+ * \brief Header defining the API for generic OQS Signature
+ */
+
+#ifndef __OQS_SIG_H
+#define __OQS_SIG_H
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include <oqs/rand.h>
+
+enum OQS_SIG_alg_name {
+	OQS_SIG_alg_default,
+};
+
+typedef struct OQS_SIG OQS_SIG;
+
+/**
+ * OQS signature object
+ */
+typedef struct OQS_SIG {
+
+	/**
+	 * PRNG
+	 */
+	OQS_RAND *rand;
+
+	/**
+	 * Specifies the name of the sig method
+	 */
+	char *method_name;
+
+	/**
+	 * Classical security in terms of the number of bits provided by the
+	 * signature method.
+	 */
+	uint16_t estimated_classical_security;
+
+	/**
+	 *  Equivalent quantum security in terms of the number of bits provided by the
+	 *  signature method.
+	 */
+	uint16_t estimated_quantum_security;
+
+	/**
+	 *  Private key length.
+	 */
+	uint16_t priv_key_len;
```

Comment:
Use size_t for lengths of buffers.  I'd also use it for the estimated_*_security members, but that's just preference.